waspc/packages/wasp-config/__tests__/mapUserSpecToAppSepcDecls.integration.test.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { describe, expect, test } from 'vitest'
+import * as AppSpec from '../src/appSpec.js'
+import {
+  makeRefParser,
+  mapApiConfig,
+  mapApiNamespace,
+  mapApp,
+  mapCrud,
+  mapJob,
+  mapOperationConfig,
+  mapPage,
+  mapRoute,
+  mapUserSpecToAppSpecDecls,
+} from '../src/mapUserSpecToAppSpecDecls.js'
+import * as Fixtures from './testFixtures.js'
+
+test('should map using mapping functions correctly', () => {
+  describe('mapUserSpecToAppSpecDecls', () => {
+    const userSpec = Fixtures.createFullUserSpec()
+    const parseEntityRef = makeRefParser('Entity', Fixtures.ALL_ENTITIES)
+    const parseRouteRef = makeRefParser('Route', Fixtures.ALL_ROUTE_NAMES)
+    const parsePageRef = makeRefParser('Page', Fixtures.ALL_PAGE_NAMES)
+
+    const result = mapUserSpecToAppSpecDecls(userSpec, Fixtures.ALL_ENTITIES)
+
+    const appDecl = getDecl(result, 'App', Fixtures.APP.FULL.NAME)
+    expect(appDecl.declValue).toStrictEqual(
+      mapApp(
+        userSpec.app.config,
+        parseEntityRef,
+        parseRouteRef,
+        userSpec.auth,
+        userSpec.server,
+        userSpec.client,
+        userSpec.db,
+        userSpec.emailSender,
+        userSpec.websocket
+      )
+    )
+    testDeclMapping({
+      declType: 'Page',
+      fixtures: Fixtures.PAGES,
+      property: 'pages',
+      mapper: mapPage,
+    })
```

Comment:
I wonder if this could be more readable / informative. So what is hapenign here. I want to check if appDecl[property] matches result of mapping with given mapping function that specific part of userSpec? Should we give it part of userSpec then, not directly Fixtures.PAGES, here?
It would be great if I could read this piece of code so I kind of undrstand what this is about.

Pseudocode:
```js
expectCorrectMapping(userSpec.pages, mapPage, getDecl(appDecl, 'Page'));
```

This is not good, I am missing some stuff here as I don't have time now to fiure it out, but I am trying to paint the idea.